DelvEdit/src/com/interrupt/dungeoneer/editor/ui/menu/generator/LevelGeneratorMenuItem.java

```diff
@@ -0,0 +1,129 @@
+                                    parent.addItem(new MenuItem(label, skin, makeLevelGeneratorAction(template)));
+                                } else {
+                                    MenuItem item = new MenuItem("No templates found", skin);
+                                    item.getLabel().setColor(0.5f, 0.5f, 0.5f, 1);
+                                    parent.addItem(item);
+                                }
+                            }
+                        } else {
+                            MenuItem item = new MenuItem("No templates found", skin);
+                            item.getLabel().setColor(0.5f, 0.5f, 0.5f, 1);
+                            parent.addItem(item);
+                        }
+
+                        menuItem.addItem(parent);
+                    }
+
+                    menuItem.addSeparator();
+                    String label = "Re-Generate Level"
+                            + (Editor.app.generatorInfo.lastGeneratedLevelTemplate != null
+                                    ? " (" + Editor.app.generatorInfo.lastGeneratedLevelTemplate.roomGeneratorType
+                                            + ", " + Editor.app.generatorInfo.lastGeneratedLevelTemplate.theme + ")"
+                                    : "");
+                    MenuItem item = new MenuItem(label, skin, makeLevelGeneratorAction())
+                            .setAccelerator(new MenuAccelerator(Keys.L, false, true));
+
+                    if (Editor.app.generatorInfo.lastGeneratedLevelTemplate == null) {
+                        item.getLabel().setColor(0.5f, 0.5f, 0.5f, 1);
+                        // TODO: Change color of accelerator label.
+                    }
+
+                    menuItem.addItem(item);
+                } else {
+                    MenuItem item = new MenuItem("No levels found", skin);
+                    item.getLabel().setColor(0.5f, 0.5f, 0.5f, 1);
+                    menuItem.addItem(item);
+                }
+            }
+
+            private ActionListener makeLevelGeneratorAction(Level template) {
+                return new ActionListener() {
+                    @Override
+                    public void actionPerformed(ActionEvent e) {
+                        if (template != null) {
+                            Level level = Editor.app.getLevel();
+
+                            if (level != null) {
+                                level.editorMarkers.clear();
+                                level.entities.clear();
+                                level.non_collidable_entities.clear();
+                                level.static_entities.clear();
+
+                                level.theme = template.theme;
+                                level.generated = true;
+                                level.dungeonLevel = 0;
+                                // TODO: This is configurable?!
```

Comment:
So many TODOs 😛

Reply:
Yes, currently I just ported the behavior that we had to the new system which only includes theme and room builder. In principle the level templates store more information, like the chunk size, level size etc.

Maybe these can be made configurable and read from the level templates as well in the future. But I'm not sure if that was something intended.